src/indexer.c

```diff
@@ -603,6 +603,20 @@ static void hash_partially(git_indexer *idx, const uint8_t *data, size_t size)
 
 static int write_at(git_indexer *idx, const void *data, off64_t offset, size_t size)
 {
+#ifdef NO_MMAP
+	size_t remaining_size = size;
+
+	/* Handle data size larger that ssize_t */
+	while (remaining_size > 0) {
+		ssize_t nb = p_pwrite(idx->pack->mwf.fd, data,
+				min(remaining_size, SSIZE_MAX),
+				offset + (size - remaining_size));
```

Comment:
Thank you! That makes it robust.

Reply:
`if (remaining_size > (size_t)(SSIZE_MAX - offset)) {
 		errno = EOVERFLOW;
 		return -1;
}`

I replaced it with the above block at the begin of function so that we catch the error early and can fail without making any changes via call to `p_pwrite`